### DISCUSSION TITLE
Qausi-steady state simulation [Porous flow]

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#19389] [Q&A Modules: General] [answered]
Hi guys,

Just wondering is it possible to set up a simulation such that, we let it run till it reaches the steady-state then we move on to the next time step.

For our mining-hydro simulation, we want to change the BCs once the simulation reaches the steady-state using `Controls`

```
[water_grad_drain1]
    type = ConditionalFunctionEnableControl
    enable_objects = 'BCs::water_grad_drain1'
    conditional_function = 'if(t>=12 & t<13,1,0)'
    execute_on = 'initial timestep_begin'
  []
  #
  [water_grad_drain2]
    type = ConditionalFunctionEnableControl
    enable_objects = 'BCs::water_grad_drain2'
    conditional_function = 'if(t>=13 & t<14,1,0)'
    execute_on = 'initial timestep_begin'
  []
.
.
```

```
  [Executioner]
    type = Transient
    start_time = 0
    end_time = 19
    dt = 1
.
.
```

I read this https://mooseframework.inl.gov/source/executioners/Transient.html (both Load steps and Quasi-Transient)
It makes sense for mechanical stimulation, but for hydro, fluid is moving with respect to time.

I check the results of my simulation, it doesn't work as I want it to, for example at T=1, we only set the BCs at the 4 walls, we expect to see pwp becomes flat by the end of that time-step, but it actually takes 10 time-steps to reach what we want.

t=1
![image](https://user-images.githubusercontent.com/79562439/141735628-e5880724-9828-439c-ae5f-ef3379458c39.png)
t=3
![image](https://user-images.githubusercontent.com/79562439/141735678-4e111f3e-8b98-4364-a605-6152c0bc1ab0.png)
t=10
![image](https://user-images.githubusercontent.com/79562439/141735714-0abea00d-f431-4d34-b6f4-d441ad3fc6c9.png)

is there any suggestion?

Kind regards,
Traiwit

**Comment by Traiwit** [comment, marked as answer]
This is the result after a steady-state run

```
  [Executioner]
    type = Steady
```

![image](https://user-images.githubusercontent.com/79562439/141751951-1839443a-b5bc-4c25-803f-9a4801470363.png)

I know I could reload and run the next steady-state simulation, but this will take time to set up the problem/read the input mesh (it's huge!) and stuff

Ideally, I'd like to do it the way I described above: reach SS > change BCs > reach SS > change BC > reach SS ...

**Reply by GiudGiud** [reply]
if you remove all the time derivative kernels, the solver will effectively be doing steady state solves every time step (with a transient executioner).
But a lot of problems need a transient to relax to the steady state. 

And maybe you need the time derivatives for the true transient with the changing BCs?

The clean way to do this is to work with Controls imo.

**Reply by Traiwit** [reply]
Hi @GiudGiud, I'm not sure how to remove all the time derivative kernels, I'm using `PorousFlowUnsaturated`, I don't see any time-term in this action. https://mooseframework.inl.gov/source/actions/PorousFlowUnsaturated.html

and yes, I'm already using Controls, to change the BCs

**Reply by GiudGiud** [reply]
ok there arent any if you are using a Steady executioner already anyway. 
In the action you would not get an option to turn them on/off anyway. 

I think using a MultiApp set up might be the move here. 
Main app: transient solve with the number of different BCs you are trying to use as the number of timesteps
Subapp: steady solve with all the BCs defined and a single one active at a time (using Controls). The subapp is a FullSolveMultiApp, that is being executed on timestep_end
Transfer: some parameter (a postprocessor for example, use a Receiver in the subapp) from main app to subapp, that is then used by the Controls in the subapp to do the switch

**Reply by Traiwit** [reply]
Hi @GiudGiud, Thanks! I will give it a go
Never touched 'MultiApp' ever before, looks very complex to me, but ill give it a try.

**Reply by Traiwit** [reply]
@GiudGiud now sure how this works, I need a porous flow expert to explain this to me but when I changed `time_unit` under `PorousFlowUnsaturated` to years (default is second), it gives me the SS result that I want

**Reply by GiudGiud** [reply]
ah makes sense, glad you figured this out.
